src/Symfony/Component/Console/Attribute/Option.php

```diff
@@ -80,28 +71,37 @@ public static function tryFrom(\ReflectionParameter $parameter): ?self
         $self->default = $parameter->getDefaultValue();
         $self->allowNull = $parameter->allowsNull();
 
-        if ('bool' === $self->typeName && $self->allowNull && \in_array($self->default, [true, false], true)) {
-            throw new LogicException(\sprintf('The option parameter "$%s" must not be nullable when it has a default boolean value.', $name));
+        if ($type instanceof \ReflectionUnionType) {
+            return self::handleUnion($self, $type);
```

Comment:
can we make it non-static instead?
```suggestion
            return $self->handleUnion($type);
```